### DISCUSSION TITLE
New LCX that would route all traffic through VPN

### DISCUSSION THREAD

**Discussion by Koliera** [tteck/Proxmox#2009] [Help Needed]
Hi, 
Have a thought/question - what would be a possible way to create a LCX that would always route ALL internet traffic through it via a provided OpenVPN/Wireguard config file?

**Comment by AlexGeek77** [comment]
I also would like to do this. I haven't started diving into doing this on my own but if a magic script happens I'd be thrilled.

**Reply by Koliera** [reply]
myeah, I guess the one way is to spin up one of the router VMs, set it to tunnel through a VPN and then link the LCX network to that. Though I'm not a 100% sure how to get that running.

**Reply by AlexGeek77** [reply]
I like the router VM idea. That way more than one LXC can connect to it as needed.

**Comment by Vakrehus** [comment]
I have an LXC running Wireguard and have added a second NIC there. The encrypted traffic runs on this second NIC. 
All containers that are to access this encrypted traffic simply receive the second NIC.

This means for example, that you have your local network for unencrypted access under eth0 and your VPN network under eth1.

You have to set up a Linux bridge in proxmox.

vmbr0 - your router
vmbr1 - your "virtual" VPN

**Reply by GameEnder** [reply]
Would you be willing to give a more involved explanation of how you set this up?
